miden-lib/src/notes/mod.rs

```diff
@@ -44,7 +44,7 @@ pub fn create_p2id_note<R: FeltRng>(
 
     let metadata = NoteMetadata::new(sender, note_type, tag, NoteExecutionHint::always(), aux)?;
     let vault = NoteAssets::new(assets)?;
-    let recipient = NoteRecipient::new(serial_num, note_script, inputs);
+    let recipient = NoteRecipient::new(serial_num, note_script.clone(), inputs);
```

Comment:
Related to the previous comments: if we return `NoteScript` from the referenced methods, we can avoid `clone()` here.